### PR TITLE
New glow options for icons

### LINKED ISSUE #1299: Further Glow Options by default
As I'm going through the custom glow library, I realized that the ability to change the speed, number of lines and length/thickness would be very nice without having to know how to write the line of code.

If possible, having the aforementioned options easily accessible to the user akin to glow color would be nice. If a low target audience, then they could all be hidden by a "Show More" tab of sorts.

Customizing them myself, which really isn't that much of a hassle

### PR DESCRIPTION
**Pull request by mrbuds**
# Description
Move glow options in a collapsible group and add more options supported by libcustomglow
Fixes #1299

demo: https://i.imgur.com/Ba9lZiI.gif (border option for pixel glow is out of the screenshot window)
new conditions: https://i.imgur.com/ScQZSOz.png

## Type of change
- [x] New feature (non-breaking change which adds functionality)

# Tests
- [x] Tested all options
- [ ] Tested all conditions => one is not working yet

# TODO:
- [ ] Condition for changing number of particules with ACShine is not working
- [ ] Update wiki

Rivers talked about using the pencil button for collapsible options, i didn't manage to use it but made it with a collapsible group, should i go back to the drawing board and get it work with pencil collapse ?
I could check how infus did it with the new text options PR.

### CODE REVIEW

**Comment by mrbuds** [comment]
Use the pencil icon for collapsing

img

**Comment by mrbuds** [comment]
it show like that with 2 icons currently
img

File: WeakAuras/RegionTypes/Icon.lua [hunk 1]
```diff
@@ -98,6 +106,75 @@ local properties = {
     setter = "SetGlowColor",
     type = "color"
   },
+  glowLines = {
+    display = { WeakAuras.newFeatureString .. L["Glow"], L["Lines & Particules"]},
```
Comment (emptyrivers) [inline WeakAuras/RegionTypes/Icon.lua]:
```suggestion
    display = { WeakAuras.newFeatureString .. L["Glow"], L["Lines & Particles"]},
```
Particules is not the English spelling.

File: WeakAurasOptions/WeakAurasOptions.lua [hunk 2]
```diff
@@ -3647,6 +3647,193 @@ function WeakAuras.PositionOptions(id, data, _, hideWidthHeight, disableSelfPoin
...
+            data.glowColor[4] * 255,
+            data.glowColor[1] * 255,
+            data.glowColor[2] * 255,
+            data.glowColor[3] * 255
+          )
+        end
+        if data.glowType == "buttonOverlay" then
+          line = ("%s %s"):format(line, color)
+        elseif data.glowType == "ACShine" then
+          line = L["%s %s, particules: %d, frequency: %0.2f, scale: %0.2f"]:format(
```
Comment (emptyrivers) [inline WeakAurasOptions/WeakAurasOptions.lua]:
```suggestion
          line = L["%s %s, particles: %d, frequency: %0.2f, scale: %0.2f"]:format(
```

File: WeakAurasOptions/WeakAurasOptions.lua [hunk 3]
```diff
@@ -3647,6 +3647,193 @@ function WeakAuras.PositionOptions(id, data, _, hideWidthHeight, disableSelfPoin
...
+      width = WeakAuras.normalWidth,
+      name = L["Color"],
+      order = order + 1.05,
+      disabled = function() return not data.useGlowColor end,
+      hidden = hiddenGlowExtra
+    },
+    glowLines = {
+      type = "range",
+      width = WeakAuras.normalWidth,
+      name = L["Lines & Particules"],
```
Comment (emptyrivers) [inline WeakAurasOptions/WeakAurasOptions.lua]:
```suggestion
      name = L["Lines & Particles"],
```

**Comment by Stanzilla** [comment]
We don't have a tooltip for the pencil atm, right?

**Comment by emptyrivers** [comment]
No, image buttons can't have tooltips.

**Comment by Stanzilla** [comment]
Hrm :/

@mrbuds you have conflicts now, sorry.

**Comment by InfusOnWoW** [comment]
I actually should implement a new image button so that we can have tooltips.

**Comment by Stanzilla** [comment]
Yes please

**Comment by InfusOnWoW** [comment]
https://pastebin.com/DBwJF4FY is the missing bit for AceOptions

**Comment by mrbuds** [comment]
Few questions:
How would you want to proceed with the AceOptions patch ? Should we wait for nev to include it upstream or we start shipping our version ?

@InfusOnWoW you talked about making glow options a subregion here https://github.com/WeakAuras/WeakAuras2/pull/1036#issuecomment-447782588 so i'm thinking to wait for the subregion PR to be merged and move this code in it's own file + use it for more regionType

**Comment by InfusOnWoW** [comment]
Conversion to a sub region should be possible, although there are likely a few small features missing for that.

**Comment by InfusOnWoW** [comment]
As to the AceOptions patch, that's only needed to get tooltips, it's not essential.

**Comment by emptyrivers** [comment]
@mrbuds you have conflicts to resolve. Outside of that, what needs to be done?

**Comment by mrbuds** [comment]
I need to find what is happening with the ACShine condition for changing number of dots not working with this aura

```
!nsvxZPUnq0Fn9H(qVtHWTT3hjG5g6W1KY60P9fLiSxS1eHKRKmeYd53E3vYMWnDY0mdJLz9Q9JZE2JyKOqakr(OR(8xM8ZcWlG54bbSxav8p6Rb8PWOmJCRgRexN()4BTktqaZYYlY207ZcRjaQNrkAJfGSl0yDRBdkRXtXH8UwBpYrGpxHM6qd5kL0tR3TZJbro9U0usx79cVaw4up)Yd)rNSYjdYxEOGQWsRvxzpAGJQwKtqLY3QLNwwAnI8jJ)Yv)2NjV68b7(cku31wjdOa66phUFwvnk2s4bFpkmDEKl1zwT1X2LLdnd4dsxi1vkJQ)TD0R(M47HEC69lAoY)1314Q9so(xgEiCI6iWyn0ZQox0H7tg9ivMvEov7LkZh37ZL5hZ)qei(6LarWPQRrhJe5rguVHxJPKI240HrUh79eg9RPYnu2GEOXECTHsfDovFuEIiF4bKbSBqPMyhGVBB0YToCN6jba3MTAfzUf16Lv90QbNGUDrNUF2uO4EOy6MIyPNtfapohkeUH4zgWCeQOHkClD1IyHFt2QBxC3QKtV2wryGN)hWIKXVzRiM(pr83HzTa(HwoyEzGXYiv6v(tea4FJJpVk(Cc)S)(J6j9xVUOy932S8R3uCbjAHwwt4Z67kwTmptanOQUHyo)Yeb8iITtjmPmSHhHd5DLYq9lLpAyQ39Mvk2Hfo8F6qt5jIa(jA680f0XyINrjMOw7JZKL5WY5zPBMWQPZGgfZDIL)hOnjk1ovDAemS0XlKZv(K8Y2Zj()nwoSMyRPcjUWslnQYhpnVNhhxyJIjlC0Gp5imBtww(fWAsVAmbI7yViImTGgPQARmk)X4kFUVthudmHsT07)pw9CWEJXHHlro8NbQ3dvBOgGMzEUlHNT29jLHUW5zkuk1Sa7NyfnfjrNJhF5HXmC(HizJheYP4DuvXkWmd6Sa7BPzDXCy38KBYC7p(73XYDhOn)ZK7)(sjmDBtc8yTd1GyzsIG24lF8sTc2VdsNIRhbWBGuzrw0D4aDPrAQXxdXW3OwV1zBrx40Lm9WaGteD0zK6)KkuMjeVb731wxfYkyKthiZtgZ5r8V)
```

**Comment by mrbuds** [comment]
With aura from last comment, how to reproduce bug:
- load game => glow show with 2 particles
- buff self to trigger condition => glow scaling seems to be increased instead of the number of particles
- open/close options while buff is still active => glow show 15 particles
- lost buff => show 2 particles
- rebuff => show 15 particles

**Comment by mrbuds** [comment]
@emptyrivers just fixed last issue

Converting to subregion, glow on aurabar, and tooltip on pencil icon can be done in another PR

**Pull request merged**